File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> mp;
        for(int num: nums) mp[num]++;
        using t = pair<int, int>;
        priority_queue<t, vector<t>, greater<t>> pq;
        for(auto x: mp) {
            pq.push({x.second, x.first});
            if(pq.size()> k) pq.pop();
        }
        vector<int>ans;
        while(!pq.empty()) {
            ans.push_back(pq.top().second); pq.pop();
        }
        return ans;
    }
};

File: Data Structures & Algorithms/maximum-product-subarray/submission-3.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        mn = mx = ans = nums[0]
        for num in nums[1:]:
            tmpMn = mn
            mn = min(num, mn*num, num*mx)
            mx = max(num, mx*num, tmpMn*num)
            ans = max(ans, mx)
        return ans
            